pkg/alertmanager/state_replication.go

```diff
@@ -142,23 +145,22 @@ func (s *state) Ready() bool {
 	return false
 }
 
-func (s *state) handleBroadcasts() {
+func (s *state) running(ctx context.Context) error {
```

Comment:
Let me leave some comments here, but not related to `running()`. Something to discuss about (but I would keep it separated from this PR):
1. Why not doing the `Settle()` in the service `starting`?
2. If we do #1, `Ready()` then would be waiting until service is running?

Thoughts @stevesg @pstibrany ?

Reply:
I like the idea of implementing `WaitReady` in terms of the waiting for the service to be running, seems like the service starting/running model fits nicely onto settling/broadcasting. Also means we can remove a the `readyc` channel, which is nice.

Reply:
Makes sense to me. I was just going to ask why do we still need `readyc` channel, good that you suggest removing it.